File: cloud/service-ribbon/src/main/java/com/chen/ribbon/serviceribbon/controller/TestController.java
package com.chen.ribbon.serviceribbon.controller;

import com.chen.ribbon.serviceribbon.service.TestService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TestController {

    @Autowired
    private TestService testService;

    @GetMapping("/test/{name}")
    public String getTest(@PathVariable String name) {
        String result = testService.hiService(name);

        return result;

    }

}

File: cas-client-test/src/main/kotlin/com/chen/spring/casclienttest/controller/TestController.kt
package com.chen.spring.casclienttest.controller

import org.springframework.security.access.annotation.Secured
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.web.bind.annotation.GetMapping


@RestController
class TestController {
    @RequestMapping("/")
    fun index(): String {
        return "访问了首页哦"
    }

    @Secured("Hello")
    @GetMapping("/hello")
    fun hello(): String {
        return "不验证哦"
    }

    @GetMapping("/auth")
    fun auth():Authentication{
        return SecurityContextHolder.getContext().authentication
    }

    @PreAuthorize("hasAuthority('TEST')")//有TEST权限的才能访问
    @GetMapping("/security")
    fun security(): String {
        return "hello world security: ${SecurityContextHolder.getContext().authentication}"
    }

    @PreAuthorize("hasAuthority('ADMIN')")//必须要有ADMIN权限的才能访问
    @GetMapping("/authorize")
    fun authorize(): String {
        return "有权限访问"
    }
}